libs/storage/Tsavorite/cs/src/core/VarLen/SpanByteExtensions.cs

```diff
@@ -13,85 +13,49 @@ public static unsafe class SpanByteExtensions
         /// <summary>
         /// Upsert with Span input
         /// </summary>
-        /// <param name="clientSession"></param>
-        /// <param name="key"></param>
-        /// <param name="desiredValue"></param>
-        /// <param name="userContext"></param>
-        /// <param name="serialNo"></param>
-        /// <returns></returns>
         public static Status Upsert<Input, Output, Context, Functions>(this ClientSession<SpanByte, SpanByte, Input, Output, Context, Functions> clientSession, ReadOnlySpan<byte> key, ReadOnlySpan<byte> desiredValue, Context userContext = default, long serialNo = 0)
             where Functions : IFunctions<SpanByte, SpanByte, Input, Output, Context>
         {
-            fixed (byte* k = key)
-            fixed (byte* v = desiredValue)
-                return clientSession.Upsert(SpanByte.FromFixedSpan(key), SpanByte.FromFixedSpan(desiredValue), userContext, serialNo);
+            fixed (byte* keyPtr = key)
```

Comment:
I don't see these Extensions used anywhere outside Tsavorite.Test, and only 10 calls in there total for Upsert and Read. This could be moved there, but probably better to change the callsites to make the proper pattern more visible

Reply:
Move the extension over to the tests or just delete it entirely try not hide the pinning?

Reply:
I think the latter is best as it makes the pinning visible and doesn't introduce a test-only layer